src/pymodaq_plugins_raspberry/daq_viewer_plugins/plugins_1D/daq_1Dviewer_daqhats.py

```diff
@@ -0,0 +1,349 @@
+import numpy as np
+from pymodaq.utils.daq_utils import ThreadCommand
+from pymodaq.utils.data import DataFromPlugins, Axis, DataToExport
+from pymodaq.control_modules.viewer_utility_classes import DAQ_Viewer_base, comon_parameters, main
+from pymodaq.utils.parameter import Parameter
+from daqhats import mcc128, OptionFlags, AnalogInputMode, TriggerModes, AnalogInputRange
+from pymodaq.utils.parameter.utils import iter_children
+
+
+class DAQ_1DViewer_daqhats(DAQ_Viewer_base):
+    """ Instrument plugin class for a 1D viewer.
+
+    This object inherits all functionalities to communicate with PyMoDAQ’s DAQ_Viewer module through inheritance via
+    DAQ_Viewer_base. It makes a bridge between the DAQ_Viewer module and the Python wrapper of a particular instrument.
+
+    Attributes:
+    -----------
+    controller: object
+        The particular object that allow the communication with the hardware, in general a python wrapper around the
+         hardware library.
+
+    """
+    params = comon_parameters + [
+        {'title': 'Trigger Active', 'name': 'trigger_active', 'type': 'bool', 'value': False},
+        {'title': 'Trigger Mode', 'name': 'trigger_mode', 'type': 'list', 'value': 'RISING_EDGE',
+         'limits': ['RISING_EDGE', 'FALLING_EDGE', 'ACTIVE_HIGH',
+                    'ACTIVE_LOW']},
+        {'title': 'External_clock', 'name': 'extclock_mode', 'type': 'bool', 'value': False},
+        {'title': 'External sampling rate', 'name': 'extclock_rate', 'type': 'int', 'value': 0},
```

Comment:
you could add here a 'visible': False option and when a user click on extclockmode, the action in the commit settings would be to show the extclock rate:
```
if param.name() == 'extclock_mode':
    self.settings.child('extclock_rate').setVisible(param.value()) 
```